modules/phase_field/src/materials/EpsAndMAndLAnisoGauss.C

```diff
@@ -0,0 +1,931 @@
+  std::vector<double> vbalibx;
+  std::vector<double> vbaliby;
+  std::vector<double> vbalibz;
+  std::vector<double> thetabalib;
+  std::vector<double> miubalibx;
+  std::vector<double> miubaliby;
+  std::vector<double> miubalibz;
+  std::vector<double> MinimaEnergy;
+  std::vector<double> MinimaL;
+
+  double Lib1;
+  double Lib2;
+  double Lib3;
+  double Lib4;
+  double Lib5;
+  double Lib6;
+  double Lib7;
+  double Lib8;
+  double Lib9;
+
+  while (!File2.eof())
+  {
+    File2 >> Lib1 >> Lib2 >> Lib3 >> Lib4 >> Lib5 >> Lib6 >> Lib7 >> Lib8 >> Lib9;
+    vbalibx.push_back(Lib1);
+    vbaliby.push_back(Lib2);
+    vbalibz.push_back(Lib3);
+    thetabalib.push_back(Lib4);
+    miubalibx.push_back(Lib5);
+    miubaliby.push_back(Lib6);
+    miubalibz.push_back(Lib7);
+    MinimaEnergy.push_back(Lib8);
+    MinimaL.push_back(Lib9);
+  }
+
+  File2.close();
+
+  //******************************* Grain quaternion outputs ****************
+
+  for (unsigned int h = 0; h < _grain_num; ++h)
+  {
+    for (unsigned int s = 0; s < _op_num; ++s)
+    {
+      if (h == op_to_grains[s])
+      {
+        _qwg[_qp] = qwR[h];
+        _qxg[_qp] = qxR[h];
+        _qyg[_qp] = qyR[h];
+        _qzg[_qp] = qzR[h];
+      }
+    }
+  }
+
+  //**************************************************************************
+
+  ADReal sumeps = 0.0;
+  ADReal sumepsx = 0.0;
+  ADReal sumepsy = 0.0;
+  ADReal sumepsz = 0.0;
+
+  ADReal sumL = 0.0;
+
+  ADReal sumepsxplus = 0.0;
+  ADReal sumepsyplus = 0.0;
+  ADReal sumepszplus = 0.0;
+
+  ADReal aniso = 0.0;
+  ADReal anisox = 0.0;
+  ADReal anisoy = 0.0;
+  ADReal anisoz = 0.0;
+
+  ADReal anisoxplus = 0.0;
+  ADReal anisoyplus = 0.0;
+  ADReal anisozplus = 0.0;
+
+  ADReal anisoL = 0.0;
+
+  ADReal Val = 0.0;
+  ADReal Val1 = 0.0;
+  ADReal sumval = 0.0;
+
+  ADReal f0saddle = 0.0;
+  ADReal g_gamma = 0.0;
+
+  ADReal eps_base = 0.0;
+  ADReal TotGaussValue = 0.0;
+  ADReal SumTotGaussValue = 0.0;
+  ADReal Totepsbar = 0.0;
+
+  ADReal eps_min = 0.0;
+  ADReal Totepsmin = 0.0;
+
+  f0saddle = 0.124961;
+
+  g_gamma = 0.471404;
+
+  // eps_base = ( (_sigmaBASE *_JtoeV * _length_scale * _length_scale) * _lgb *
+  // (std::sqrt(f0saddle)) ) / (g_gamma) ;
+  eps_base = (3.0 / 4.0) * (_sigmaBASE * _JtoeV * _length_scale * _length_scale) * _lgb;
+
+  //********************** sum (etai*etaj)^2 at each quadrature point **************************
+
+  for (unsigned int m = 0; m < _op_num - 1; ++m)
+  {
+    for (unsigned int n = m + 1; n < _op_num; ++n) // m<n
+    {
+      Val1 = (((*_vals[m])[_qp]) * ((*_vals[m])[_qp])) * (((*_vals[n])[_qp]) * ((*_vals[n])[_qp]));
+      sumval += Val1;
+    }
+  }
+
+  //********************** Main anisotopy calculation **************
+
+  for (unsigned int b = 0; b < _grain_num - 1; ++b)
+  {
+
+    for (unsigned int m = 0; m < _op_num - 1; ++m)
+    {
+
+      for (unsigned int a = b + 1; a < _grain_num; ++a)
+      {
+
+        for (unsigned int n = m + 1; n < _op_num; ++n)
+        {
```

Comment:
```suggestion
  for (unsigned int b = 0; b < _grain_num - 1; ++b)
    for (unsigned int m = 0; m < _op_num - 1; ++m)
      for (unsigned int a = b + 1; a < _grain_num; ++a)
        for (unsigned int n = m + 1; n < _op_num; ++n)
```
please avoid unnecessary braces and empty lines. Also take a look at our code base and see how we use `make_range` in loops in our newer code.

Reply:
That was fixed. See updated files.